content/manager/manager-upgrade.md

```diff
@@ -0,0 +1,219 @@
+* The Manager file-system must hold the input properties used by the previous installation/upgrade under `/opt/cloudify/<NODE_NAME>/node_proprties/properties.json` to allow rollback.
+
+### Node Specific Validations
+
+* Elasticsearch
+    * `es_heap_size` property must not be smaller than current memory size.
+    * Elasticsearch must be available on the specified port.
+    * Elasticsearch must hold a `provider_context` index.
+* Rabbitmq
+    * Broker configuration may not change. Specifically:
+      `broker_user`, `broker_pass`, `broker_ip`, `broker_ssl_enabled` and `broker_ssl_cert`
+* Manager Config
+    * SSL communication configuration may not change. Specifically `ssl`, `security` and `ssh_user`
+
+## Upgrade
+
+Once the new [simple-manager-blueprint](https://github.com/cloudify-cosmo/cloudify-manager-blueprints) is configured along with all of the required resources,
+the Manager upgrade process may begin. The upgrade process will use `install` workflow to preform the Manager upgrade process.
+This workflow can be divided into three main steps as can also be seen in the diagram above:
+
+1. Create Upgrade Snapshot - The upgrade snapshot will hold all of elasticsearch and influxdb data prior to the upgrade. A failure in this part of the upgrade process will not require running the `cfy rollback` command.
+2. Upgrade Services - Upgrade the Manager services using the new sources and resources.
+3. Restore Upgrade Snapshot - Once all services have been installed and configured and the Cloudify Manager is up, the upgrade snapshot will be restored to the new Manager.
+
+A failure in any of these steps can either be handled by running the `cfy upgrade` command again, or by running the `cfy rollback` command as addressed later in the document.
+
+{{% gsNote title="Note" %}}
+The upgrade scripts are designed to be idempotent and so there is no risk of running the `cfy upgrade` command consecutively.
+This can be useful in-case of failures due to power/network issues or in case of failures related to the new input properties provided.
+{{% /gsNote %}}
+
+
+### Upgrade Execution
+Starting the upgrade process requires having the Cloudify Manager in [maintenance-mode](http://??). This will prevent having running executions during the upgrade process.
+To activate maintenance-mode, run:
+{{< gsHighlight  bash  >}}
+cfy maintenance-mode activate
+{{< /gsHighlight >}}
+The Cloudify Manager upgrade will be possible only once the manager enters maintenance mode.
+Once in maintenance-mode, using the pre-configured [simple-manager-blueprint-inputs.yaml](https://github.com/cloudify-cosmo/cloudify-manager-blueprints/blob/3.3.1/simple-manager-blueprint-inputs.yaml)
+execute the following CLI command to initiate the upgrade process:
+{{< gsHighlight  bash  >}}
+cfy upgrade --blueprint-path simple-manager-blueprint.yaml --inputs simple-manager-blueprint-inputs.yaml -v
```

Comment:
Should this suggest `--install-plugins` as that's almost definitely going to be required on an upgrade?